<?xml version="1.0" ?>
<launch>
  <!-- Robot gains setup and test arguments -->
  <arg name="show_error" default="false" />
  <arg name="gains" default="false" />
  <arg name="rviz" default="false" />

  <arg name="robot_ip" />
  <arg name="load_gripper" default="true" />
  <include file="$(find franka_control)/launch/franka_control.launch" >
    <arg name="robot_ip" value="$(arg robot_ip)" />
    <arg name="load_gripper" value="$(arg load_gripper)" />
  </include>

  <rosparam command="load" file="$(find panda_3dbioprint_real)/config/panda_control.yaml" />
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="cartesian_impedance_controller"/>

  <node if="$(arg rviz)" pkg="rviz" type="rviz" output="screen" name="rviz"/>
  <node if="$(arg gains)" name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" required="false" />
  <node if="$(arg show_error)" name="controller_error_plot" pkg="panda_3dbioprint_real" type="controller_error_plot.py" />

</launch>